src/Models/NonhydrostaticModels/pressure_correction.jl

```diff
@@ -1,19 +1,25 @@
 import Oceananigans.TimeSteppers: calculate_pressure_correction!, pressure_correct_velocities!
 
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: step_free_surface!
+
 """
     calculate_pressure_correction!(model::NonhydrostaticModel, Δt)
 
 Calculate the (nonhydrostatic) pressure correction associated `tendencies`, `velocities`, and step size `Δt`.
 """
 function calculate_pressure_correction!(model::NonhydrostaticModel, Δt)
 
+    if !isnothing(model.free_surface)
+        step_free_surface!(model.free_surface, model, model.timestepper, Δt)
+        # "First" barotropic pressure correction
+        pressure_correct_velocities!(model, model.free_surface, Δt)
+    end
```

Comment:
@shriyafruitwala

see here, a check is added that steps forward the free surface. You should change the lines in this if-statement according to what you need.